app/src/main/java/com/yacgroup/yacguide/BaseNavigationActivity.kt

```diff
@@ -28,21 +28,25 @@ import androidx.appcompat.app.ActionBarDrawerToggle
 import androidx.appcompat.widget.Toolbar
 import androidx.core.view.GravityCompat
 import androidx.drawerlayout.widget.DrawerLayout
+import androidx.viewbinding.ViewBinding
 import com.google.android.material.navigation.NavigationView
 
-abstract class BaseNavigationActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {
+abstract class BaseNavigationActivity<ViewBindingType: ViewBinding> : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {
 
-    abstract fun getLayoutId(): Int
+    protected lateinit var binding: ViewBindingType
```

Comment:
I am not glad with this name since it does not tell anything about its purpose. `viewBinding` or `uiBinding` might be a better name.

Reply:
I agree. I going to name it `viewBinding`.